blocks/product-recommendations/product-recommendations.js

```diff
@@ -114,6 +118,20 @@ export default async function decorate(block) {
     // Get purchase history
     context.userPurchaseHistory = getPurchaseHistory(storeViewCode);
 
+    let recommendationsData = null;
+
+    // Get data from the event bus to set publish events
+    events.on(
+      'recommendations/data',
+      (data) => {
+        recommendationsData = data;
+        if (data?.items?.length) {
+          recommendationsData = data;
```

Comment:
Does the slot (Footer in this case) not provide the current context? If it did, you could just reference `ctx.data.items` perhaps, and filter over that when trying to create the `recommendationsUnit`.